mmv1/templates/terraform/examples/bigquery_connection_kms.tf.erb

```diff
@@ -0,0 +1,61 @@
+resource "google_sql_database_instance" "instance" {
+    name             = "<%= ctx[:vars]['database_instance_name'] %>"
+    database_version = "POSTGRES_11"
+    region           = "us-central1"
+    settings {
+		tier = "db-f1-micro"
+	}
+
+    deletion_protection  = "<%= ctx[:vars]['deletion_protection'] %>"
+}
+
+resource "google_sql_database" "db" {
+    instance = google_sql_database_instance.instance.name
+    name     = "db"
+}
+
+resource "random_password" "pwd" {
+    length = 16
+    special = false
+}
+
+resource "google_sql_user" "user" {
+    name = "<%= ctx[:vars]['username'] %>"
+    instance = google_sql_database_instance.instance.name
+    password = random_password.pwd.result
+}
+
+resource "google_bigquery_connection" "<%= ctx[:primary_resource_id] %>" {
+    friendly_name = "👋"
+    description   = "a riveting description"
+    location      = "US"
+    kms_key_name  = google_kms_crypto_key.default.id
+    cloud_sql {
+        instance_id = google_sql_database_instance.instance.connection_name
+        database    = google_sql_database.db.name
+        type        = "POSTGRES"
+        credential {
+          username = google_sql_user.user.name
+          password = google_sql_user.user.password
+        }
+    }
+}
+
+resource "google_kms_key_ring" "default" {
+    name     = "<%= ctx[:vars]['kms_key_ring'] %>"
+    location = "us"
+}
+
+resource "google_kms_crypto_key" "default" {
+    name     = "<%= ctx[:vars]['kms_key_name'] %>"
+    key_ring = google_kms_key_ring.default.id
+}
+
+data "google_project" "project" {
+}
+
+resource "google_kms_crypto_key_iam_member" "default" {
+    crypto_key_id = google_kms_crypto_key.default.id
+    member        = "serviceAccount:bq-${data.google_project.project.number}@bigquery-encryption.iam.gserviceaccount.com"
+    role          = "roles/cloudkms.cryptoKeyEncrypterDecrypter"
+}
```

Comment:
Since this is modifying a project-wide service account, it should also be bootstrapped using BootstrapPSARole similar to:
https://github.com/GoogleCloudPlatform/magic-modules/blob/07c06ecd2a0d5b9a5d040102eb4dc26676e6152b/mmv1/products/cloudfunctions2/Function.yaml#L129-L132

(It doesn't need to be used in the template, just present in the test_vars_overrides.)

Reply:
Specifically it would be:

```yaml
policyChanged: 
         "acctest.BootstrapPSARole(t, \"bq-\", \"bigquery-encryption\", 
         \"roles/cloudkms.cryptoKeyEncrypterDecrypter\"\ 
         )" 
```

Reply:
Added!